### PR TITLE
fix: fix keycloak dev realm

### PR DESCRIPTION
**Pull request by PaddseL**
_What issues does this PR close?_
The Keycloak dev realm is out of sync with the local E2E configuration. This PR configures the Keycloak dev realm, so that it works with the local E2E development setup and the User Switcher of the frontend.

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md and linked the changes to their issues. See [changelog_update.md](https://github.com/sovity/authority-portal/blob/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: authority-portal-backend/docs/dev/keycloak-realm-export/README.md [hunk 1]
```diff
@@ -13,6 +13,15 @@ To run this script, execute the following command from the backend project's roo
 ./docs/dev/keycloak-realm-export/extract-realm.sh
 ```
 
+After the script has run, the exported [realm.json](../../../authority-portal-quarkus/src/main/resources/realm.json) may contain unrecognized fields which leads to an import error on next startup.
+Quarkus indicates those with an error message on startup like this:
```
Comment (Linkshegelianer) [inline authority-portal-backend/docs/dev/keycloak-realm-export/README.md:17]: I suppose this issue is fixed by the newest version or cheap workarounds depending on the exact property causing an error: https://stackoverflow.com/questions/78575033/quarkus-and-keycloak-compatibility

The question is, how long will it take to troubleshoot it and do we need to do this?

File: authority-portal-backend/docs/dev/keycloak-realm-export/README.md [hunk 2]
```diff
@@ -13,6 +13,15 @@ To run this script, execute the following command from the backend project's roo
...
+
+```
+2024-12-20 10:08:44,484 ERROR [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (build-57) Realm realm.dev.json resource can not be opened: Unrecognized field "firstBrokerLoginFlow" (class org.keycloak.representations.idm.RealmRepresentation), not marked as ignorable [...]
+```
+
+To fix this, remove the unrecognized fields from the exported [realm.json](../../../authority-portal-quarkus/src/main/resources/realm.json) file.
```
Comment (Linkshegelianer) [inline authority-portal-backend/docs/dev/keycloak-realm-export/README.md:23]: Remove or comment out? Not clear from this docs

File: authority-portal-backend/authority-portal-quarkus/src/main/resources/realm.dev.json [hunk 3]
```diff
@@ -2923,16 +2954,24 @@
...
+    "cibaExpiresIn": "120",
+    "oauth2DeviceCodeLifespan": "600",
     "parRequestUriLifespan": "60",
-    "cibaInterval": "5"
+    "clientSessionMaxLifespan": "0",
+    "frontendUrl": "",
+    "acr.loa.map": "{}"
   },
-  "keycloakVersion": "22.0.1",
+  "keycloakVersion": "24.0.4",
```
[the remark this answers is not in the document]
Comment (Linkshegelianer) [reply authority-portal-backend/authority-portal-quarkus/src/main/resources/realm.dev.json:2974]: My bad, this is just for dev

[the remark this answers is not in the document]
Comment (PaddseL) [reply authority-portal-backend/authority-portal-quarkus/src/main/resources/realm.dev.json:2974]: I guess this is just the version that the file itself was exported with.

**Review by Linkshegelianer — APPROVED** [review]
LGTM

**Pull request merged**